README.md

```diff
@@ -5,40 +5,157 @@ Please edit this README.md file with your answers to the worksheet questions.
+|true |true |false        |false         |
 
 ### c
+|A    |B    |C    |(A and B) or (A and C)|A and (B or C)|
+|-----|-----|-----|----------------------|--------------|
+|false|false|false|false                 |false         |
+|false|false|true |false                 |false         |
+|false|true |false|false                 |false         |
+|false|true |true |false                 |false         |
+|true |false|false|false                 |false         |
+|true |false|true |true                  |true          |
+|true |true |false|true                  |true          |
+|true |true |true |true                  |true          |
+
 
 ### d
+|A    |B    |C    |(A or B) and (A or C)|A or (B and C)|
+|-----|-----|-----|---------------------|--------------|
+|false|false|false|false                |false         |
+|false|false|true |false                |false         |
+|false|true |false|false                |false         |
+|false|true |true |true                 |true          |
+|true |false|false|true                 |true          |
+|true |false|true |true                 |true          |
+|true |true |false|true                 |true          |
+|true |true |true |true                 |true          |
+
 
 ## Question 4
 
 ### a
+|file_exists("a.txt")|file_exists("b.txt")|first program|second program|
+|--------------------|--------------------|-------------|--------------|
+|FALSE               |FALSE               |TRUE         |TRUE          |
+|FALSE               |TRUE                |TRUE         |TRUE          |
+|TRUE                |FALSE               |TRUE         |TRUE          |
+|TRUE                |TRUE                |FALSE        |FALSE         |
+
 
 ### b
+|type(x) == int|x > 7|type(x) == float|first program|second program|
+|--------------|-----|----------------|-------------|--------------|
+|FALSE         |FALSE|FALSE           |FALSE        |FALSE         |
+|FALSE         |FALSE|TRUE            |FALSE        |FALSE         |
+|FALSE         |TRUE |FALSE           |FALSE        |FALSE         |
+|FALSE         |TRUE |TRUE            |TRUE         |TRUE          |
+|TRUE          |FALSE|FALSE           |FALSE        |FALSE         |
+|TRUE          |FALSE|TRUE            |FALSE        |FALSE         |
+|TRUE          |TRUE |FALSE           |TRUE         |TRUE          |
+|TRUE          |TRUE |TRUE            |TRUE         |TRUE          |
+
 
 ### c
+|X == 0|Y == 0|first program|second program|
+|------|------|-------------|--------------|
+|TRUE  |TRUE  |FALSE        |FALSE         |
+|TRUE  |FALSE |FALSE        |FALSE         |
+|FALSE |TRUE  |FALSE        |FALSE         |
+|FALSE |FALSE |TRUE         |TRUE          |
 
 ### d
+|X > 10|Y > 0|first program|second program|
+|------|-----|-------------|--------------|
+|FALSE |FALSE|TRUE         |FALSE         |
+|FALSE |TRUE |TRUE         |FALSE         |
+|TRUE  |FALSE|FALSE        |FALSE         |
+|TRUE  |TRUE |TRUE         |TRUE          |
```

Comment:
I think you have the right idea with all of these -- however it would be good to see some more explanation of how you arrived at the "first program" and "second program" truth values and what they represent in terms of the behaviour of the program.